### ISSUE TITLE
Block properties ignored

### ISSUE THREAD

**Issue by LemADEC** [pWn3d1337/Techguns#70]
As of Techguns.beta.1.2_alphatest3.jar, all placed explosives break blocks without taking into account the block properties like hardness and explosion resistance.
In other words, the mod doesn't integrate with other mods providing bunker/hull protection blocks.

**Comment by pWn3d1337** [comment]
Yes the charges do break all blocks except unbreakable ones. I will change this so the normal ones can't break obsidian, and the advanced ones can break obsidian, but not stronger things.

**Comment by pWn3d1337** [comment]
I have implemented a solution. The explosive charges now take the block hardness as limit. Both max hardness values (for normal and avanced charges) can be set in config file. by default it will be 30.0 and 100.0.

Also the charges will post a blockBreakevent and not break when it gets cancelled. The explosion now also moves in a line and stops when a block in that line can't get broken. The block break event gets the owning player from the tileEntity, the owner is set when placing.

**Comment by LemADEC** [comment]
So far, it's looking good, thanks for the fix!
